Code/Framework/AzCore/Tests/AZStd/TypeTraits.cpp

```diff
@@ -290,198 +290,317 @@ namespace UnitTest
+        static_assert(AZStd::is_same_v<AZStd::iter_difference_t<UnitTest::IteratorTraitsSpecializationTest>, ptrdiff_t>);
+    }
+    TEST(TypeTraits, IterReferenceCompiles)
     {
-        constexpr operator int() { return {}; };
+        static_assert(AZStd::is_same_v<AZStd::iter_reference_t<typename AZStd::string_view::iterator>, const char&>);
+        static_assert(AZStd::is_same_v<AZStd::iter_reference_t<typename AZStd::list<char>::iterator>, char&>);
+        static_assert(AZStd::is_same_v<AZStd::iter_reference_t<AZ::u32[]>, AZ::u32&>);
+        static_assert(AZStd::is_same_v<AZStd::iter_reference_t<AZ::u32*>, AZ::u32&>);
+        static_assert(AZStd::is_same_v<AZStd::iter_reference_t<typename AZStd::list<char>::const_iterator>, const char&>);
+        static_assert(AZStd::is_same_v<AZStd::iter_reference_t<std::istreambuf_iterator<char>>, char>);
+    }
+
+    // Test customizing the ranges::iter_move function
+    struct IterMoveCustomizationPointTest
+    {
+        int operator*();
     };
-    static_assert(NumericCastInvocable<AzNumericCastConvertibleCompileTest>, "aznumeric_cast conversion operator overload is should be compilable");
+    char&& iter_move(IterMoveCustomizationPointTest&);
+    TEST(TypeTraits, IterRvalueReferenceCompiles)
+    {
+        static_assert(AZStd::is_same_v<AZStd::iter_rvalue_reference_t<typename AZStd::string_view::iterator>, const char&&>);
+        static_assert(AZStd::is_same_v<AZStd::iter_rvalue_reference_t<typename AZStd::list<char>::iterator>, char&&>);
+        static_assert(AZStd::is_same_v<AZStd::iter_rvalue_reference_t<AZ::u32[]>, AZ::u32&&>);
+        static_assert(AZStd::is_same_v<AZStd::iter_rvalue_reference_t<AZ::u32*>, AZ::u32&&>);
+        static_assert(AZStd::is_same_v<AZStd::iter_rvalue_reference_t<typename AZStd::list<char>::const_iterator>, const char&&>);
+        static_assert(AZStd::is_same_v<AZStd::iter_rvalue_reference_t<std::istreambuf_iterator<char>>, char>);
+        static_assert(AZStd::is_same_v<AZStd::iter_rvalue_reference_t<IterMoveCustomizationPointTest>, char&&>);
+    }
+
+    // VS2017 workaround, calling decltype directly on the fully specialized aznumeric_cast template
+    // function fails with error C3556:  'aznumeric_cast': incorrect argument to 'decltype'
+    // So invoke the attempt to invoke function in a non-evaluated context and SFINAE to prevent a compile
+    // error
+    template <typename T, typename = void>
+    constexpr bool NumericCastInvocable = false;
+    template <typename T>
+    constexpr bool NumericCastInvocable<T, AZStd::void_t<decltype(aznumeric_cast<int>(AZStd::declval<T>()))>> = true;
+    TEST(TypeTraits, NumericCastConversionOperatorCompiles)
+    {
+        struct AzNumericCastConvertibleCompileTest
+        {
+            constexpr operator int() { return {}; };
+        };
+        static_assert(NumericCastInvocable<AzNumericCastConvertibleCompileTest>, "aznumeric_cast conversion operator overload is should be compilable");
```

Comment:
"is should" -> "should"

Reply:
This section of the code didn't change, but I'll fix the typo anyway.